txscript/standard.go

```diff
@@ -350,6 +350,21 @@ func PayToAddrScript(addr btcutil.Address) ([]byte, error) {
 	return nil, ErrUnsupportedAddress
 }
 
+// NullDataScript creates a provably prunable script
+// containing OP_RETURN followed by the passed data.
+func NullDataScript(data []byte) ([]byte, error) {
+	if len(data) > MaxDataCarrierSize {
+		return nil, ErrStackLongScript
+	}
+
+	script := make([]byte, 3+len(data))
```

Comment:
This will produce OP_RETURN scripts with non-canonical data pushes that would end up being rejected by the majority of the network.

Please use this instead:

````Go
	builder := txscript.NewScriptBuilder()
	return builder.AddOp(txscript.OP_RETURN).AddData(data).Script()
```